eos-update-flatpak-repos

```diff
@@ -447,6 +448,16 @@ def _filter_refs(refs, name=None, prefix=None, kind=None, arch=None,
     return ret
 
 
+def _filter_ostree_refs(refs, kind, name='*', prefix=None, arch='*',
+                        branch='*', origin='*'):
+    if prefix:
+        name = '{}*'.format(prefix)
+
+    pattern = '{}:{}/{}/{}/{}'.format(origin, kind.value_nick, name, arch, branch)
+
+    return fnmatch.filter(refs, pattern)
```

Comment:
The worrier in me is worrying about the possibility of this ever matching and including `/ostree/*/*` (deployment refs). Similarly for the other `_filter_refs()` method.

Reply:
This... seems intensely unlikely...? In as much as you can rely on anything in Python, provided kind is a FlafpakRefKind, then kind.value_nick can only ever be 'app' or 'runtime', and the /ostree/* refs do not have origin: prefixes. The other refs are FlatpakRefs so examine installed apps and their deploy files.

Reply:
Maybe something like `assert origin != ''` and `assert kind.value_nick in ['app', 'runtime']`. I may be being over-paranoid, but I’d prefer that over this script silently deleting an important ref in future if something changes in flatpak.